Model/DutyCalculator.php

```diff
@@ -288,7 +290,6 @@ public function fillOutQuoteAndSessionUsingFeeReturned($duty, $address, $apply,
             $this->checkoutSession->setBaseReachDuty(0);
             $this->checkoutSession->setReachDuty($duty_adjusted);
             $quote->setDhlQuoteId('');
-            $this->checkoutSession->setDhlQuoteId('');
```

Comment:
if the session isn't getting cleared when the checkbox isn't selected, then couldn't a customer 

- go through the checkout flow
- click the duties and taxes checkbox (the session gets a quoteId), 
- leave the checkout flow then when they 
- go through checkout again they 
- don't click the checkbox/deselect checkbox, 
- wouldn't their session still have the old quoteId?

Just want to double check that removing the clearing of the quoteId in the session doesn't result in issues like the one I outlined above

Reply:
Few things:

Some values are not getting cleared from session (deliberately). This is to be able to reuse the D&T (as well as DHLQuoteid) values without a new DHL call (if nothing else changes except the checkbox selection ).  
What is getting into the database is from Quote not from Session.
So when user is deselecting the apply D&T checkbox - the info related to D&T is getting cleared from the Quote but not from Session. This is so that just in case the user select the apply D&T option again we can reload value from session to quote without making a new DHL call. If we clear the DHLQuoteID from session then if the user changes mind during the same session then we would not have the DHLquoteid without making a new call.
However there are cases when it would be a problem (like as listed in MAG-115). This is the reason I was requesting (please have a look at `dev-magento` slack channel ) last Friday to bump up the priority of MAG-115 so that more event handlers (like MAG-81) are added for proper handling of such cases. However that did not happen (I am going to share the link with you shortly). Thanks!

Reply:
ohh thank you! I appreciate the time you spent clarifying this